src/main/osd/osd_elements.c

```diff
@@ -603,11 +588,8 @@ static void osdElementCrashFlipArrow(osdElementParms_t *element)
                 }
             }
         }
-    } else {
-        element->buff[0] = ' ';
+        element->buff[1] = '\0';
```

Comment:
Should be `element->buff[0] = '\0'`. Or, to use the commonly used pattern, introduce `unsigned pos = 0`, and then assign `element->buff[pos++]` above, and then `element->buff[pos] = '\0'` at the end.

Reply:
I don't think so. Now what it does is write an arrow + `\0` when in flip over after crash, and does nothing when not in this mode. I suppose that the buffer has been initialized to `\0`.
I have observed other elements, like this:

https://github.com/betaflight/betaflight/blob/b3b535724e6be5de40efa2c9bc3ab03bf5ebe74e/src/main/osd/osd_elements.c#L493-L498

I have not touched this function, but this element does not add a `\0` at the end, so I suppose all the buffer has been initialized with this value. If not this will fail. Am I right? @etracer65 you refactored completely the OSD code, what is the correct way to do this?

Reply:
`strcpy` copies the terminating null byte. Unless we know the buffer will always be initialised to `\0` we need to add the terminator.

Reply:
Yes, but if it does not enter in the if, it will not add a `\0` to the buffer. Is the same case than in the flip arrow. If it is not the correct way I can add the fix to both elements.